### PR TITLE
SERVER-80231: Add conditional statement for variable declaration

### PR DESCRIPTION
**Pull request by siddux**
The changes solve the minor issue reported [here](https://jira.mongodb.org/browse/CXX-2708).

### CODE REVIEW

File: src/mongo/util/net/sock.h [hunk 1]
```diff
@@ -73,7 +73,9 @@ extern const int portRecvFlags;
 inline void closesocket(int s) {
     close(s);
 }
-const int INVALID_SOCKET = -1;
+#ifndef INVALID_SCOKET
```
Comment (suenalaba) [inline src/mongo/util/net/sock.h:76]:
```suggestion
#ifndef INVALID_SOCKET
```

spelling error

File: src/mongo/util/net/sock.h [hunk 2]
```diff
@@ -73,7 +73,9 @@ extern const int portRecvFlags;
...
+#define INVALID_SOCKET = -1;
+#endif
 typedef int SOCKET;
```
Comment (suenalaba) [inline src/mongo/util/net/sock.h:79]:
```suggestion
#endif

typedef int SOCKET;
```

nit: good to add a linebreak, for readability here

**Review by suenalaba — CHANGES_REQUESTED** [review]
good work, some comments

**Comment by siddux** [comment]
Thanks for your comments

**Comment by kelly-cs** [comment]
Thanks for making this pull request! We just need you to create a new SERVER ticket on [JIRA](https://jira.mongodb.org/) to help us track the PR. Linking https://jira.mongodb.org/browse/CXX-2708 will be great too.

**Comment by kelly-cs** [comment]
We're changed the existing ticket to [SERVER-80231](https://jira.mongodb.org/browse/SERVER-80231) (originally [CXX-2708](https://jira.mongodb.org/browse/CXX-2708)) to track this work, so no need to make a new ticket.

**Comment by siddux** [comment]
> > Thanks for your comments
> 
> Please fix he typos and recommit

Just fixed.

**Review by suenalaba — APPROVED** [review]
LGTM.

**Comment by patrickfreed** [comment]
The mongo codebase isn't intended to be depended upon as a library externally, and there shouldn't be multiple definitions of INVALID_SOCKET internal to mongo. In light of that, I'm going to close this, but thanks for the submission anyways!

**Pull request closed without merging**